docs/REFERENCE.md

```diff
@@ -123,39 +165,89 @@ const errResult = Err.of(new UserNotFoundError());
+const sameErrResult = errResult.map(err => 123); 
+```
 
 ---
 
 ### err.mapErr(ErrorClass, callback)
 
-Map `Err` result of specified class to a different result.
+Map the `Err` result of specified class to a different result.
 Interface common for both types of results: [result.mapErr(ErrorClass, callback)](#resultmaperrerrorclass-callback)
 
+**Examples:**
+
+```ts
+import { Ok } from 'type-safe-errors';
+import { UserNotFoundError, Http404Error } from './errors';
+
+const errResult = Err.of(new UserNotFoundError());
+
+// Ok.of('User not found')
+const sameErrResult = errResult.mapErr(UserNotFoundError, err => 'User not found'); 
+
+// Err result of Http404Error is returned
+const httpErrResult = errResult.mapErr(UserNotFoundError, err => Err.of(new Http404Error())); 
+```
+
 ---
 
 ### err.mapAnyErr(callback)
 
-Map `Err` result to a different result.
+Map the `Err` result to a different result.
 Interface common for both types of results: [result.mapAnyErr(callback)](#resultmapanyerrcallback)
 
+**Examples:**
+
+```ts
+import { Ok } from 'type-safe-errors';
+import { UserNotFoundError, Http404Error } from './errors';
+
+const errResult = Err.of(new UserNotFoundError());
+
+// Ok.of('User not found')
+const sameErrResult = errResult.mapAnyErr(err => 'User not found'); 
+
+// Err result of Http404Error is returned
+const httpErrResult = errResult.mapAnyErr(err => Err.of(new Http404Error())); 
+```
 ---
 
 ### err.unsafePromise()
 
-Map `Err` to a rejected [promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise). 
-Interface common for both types of results: [result.unsafePromise()](#resultunsafePromise)
+Map an `Err` result to a rejected [promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise). This is a common interface for both `Ok` and `Err` results: [result.unsafePromise()](#resultunsafePromise).
+
+**Examples:**
+
+```typescript
+import { Err } from 'type-safe-errors';
+import { UserNotFoundError } from './errors';
+
+const errResult = Err.of(new UserNotFoundError());
+// Promise rejects with value of UserNotFoundError error instance
+const rejectedPromise = errResult.unsafePromise();
+
+```
 
 ---
 
 ## Result
 
-`Result` provide static utility functions to work with multiple results.
+`Result` provides static utility functions to work with multiple results.
 
 ### Result.combine([result1, result2, ...])
 
-Combine provided results list into single result. If all provided results are `Ok`, returned result will be `Ok` of array of provided results values: `[Ok<A>, Ok<B>] -> Ok<[A, B]>`  
+Combines a list of provided results into a single result. If all provided results are `Ok`, the returned result will be an `Ok` containing an array of values from the provided results: `[Ok<A>, Ok<B>] -> Ok<[A, B]>`.  
 If provided results list have at least one `Err` result, returned result will be `Err` of first `Err` result value found in the array.
```

Comment:
Is that still true isn't that sum of error types ?

ex: 
```ts
const ok1Result = Ok.of(5);
const ok2Result = Ok.of(9);
const err1Result = Err.of(new UserNotFoundError());
const err2Result = Err.of(new UserBlocked());
const okSumResult = Result.combine([ok1Result, err1Result , ok2Result, err2Result])
```
` type of: okSumResult === Result<5 | 9, UserNotFoundError | UserBlocked> ?`

Reply:
its not exactly like that. there is better example to consider:
```ts
const ok1Result = Ok.of(1);
const ok2Result = Ok.of(2);
const maybeErr1Result = Ok.of(5).map(() => Math.random() > 0.5 ? Err.of(new Error1()) : 5);
const maybeErr2Result = Ok.of(6).map(() => Math.random() > 0.5 ? Err.of(new Error2()) : 6);
const okSumResult = Result.combine([ok1Result, maybeErr1Result , ok2Result, maybeErr2Result])
```
`type of: okSumResult === Result<[1, 5, 2, 6], Error1 | Error2>`.

I think you confuse types with execution. The description saying:
> If provided results list have at least one `Err` result, returned result will be `Err` of first `Err` result value found in the array.

In execution it will be first error which will happen. But in types you do not know which error will happen first, so you have potentially all errors available.

Reply:
maybe if that was not clear for me we can change to your example because it's more descriptive?

Reply:
I improved combine signature and add more examples.